qiime2/core/type/signature.py

```diff
@@ -268,7 +268,10 @@ def _assert_valid_outputs(self, outputs):
     def decode_parameters(self, **kwargs):
         params = {}
         for key, spec in self.parameters.items():
-            params[key] = spec.qiime_type.decode(kwargs[key])
+            if spec.has_default() and spec.default is kwargs[key] is None:
```

Comment:
Woah, I've never seen this form of conditional before (`<something> is <something> is <something>`). I played around with it locally and it seems to work... maybe it's clearer to make this a more "standard" conditional to help with readability?

```python
if spec.has_default() and spec.default is None and kwargs[key] is None:
    ...
```